src/plugins/intel_npu/src/plugin/src/compiled_model.cpp

```diff
@@ -128,17 +135,23 @@ std::shared_ptr<ov::ISyncInferRequest> CompiledModel::create_sync_infer_request(
 }
 
 void CompiledModel::export_model(std::ostream& stream) const {
-    const auto& blob = _networkPtr->compiledNetwork;
-    stream.write(reinterpret_cast<const char*>(blob.data()), blob.size());
+    std::stringstream str;
+    if (_networkPtr->compiledNetwork.size() == 0 && _networkPtr->metadata.graphHandle != nullptr) {
+        const auto blob = _compiler->getCompiledNetwork(_networkPtr);
+        stream.write(reinterpret_cast<const char*>(blob.data()), blob.size());
+        str << "Blob size: " << blob.size() << ", hash: " << std::hex << hash(blob);
+    } else {
+        const auto& blob = _networkPtr->compiledNetwork;
+        stream.write(reinterpret_cast<const char*>(blob.data()), blob.size());
+        str << "Blob size: " << blob.size() << ", hash: " << std::hex << hash(blob);
```

Comment:
Anyway, this can be out of the if condition, right?

Reply:
yes, i have move this part out in the latest commit